app/src/main/java/com/example/taskyy/domain/navigation/Navigation.kt

```diff
@@ -72,40 +55,38 @@ fun Nav() {
             }
             composable(route = Screen.Register.route) {
                 val registerViewModel = hiltViewModel<RegisterViewModel>()
+                val state = registerViewModel.state
                 ObserveAsEvents(registerViewModel.events) { event ->
                     when (event) {
-                        is RegisterEvent.RegistrationSuccessful -> navController.navigate(Screen.Agenda.route)
+                        is RegisterEvent.RegistrationSuccessful -> navController.navigate(Screen.Login.route)
                         else -> {}
                     }
                 }
-                val state = registerViewModel.state
                 RegisterScreen(
                     state = state,
-                    onEvent = { event ->
-                        when (event) {
-                            is RegisterEvent.OnGetStartedClick -> if (state.isRegistrationSuccessful)
-                                navController.navigate(Screen.Agenda.route)
-
-                            else -> {}
-                        }
-                        registerViewModel.onEvent(event)
-                    }
+                    onEvent = { registerViewModel.onEvent(it) }
                 )
             }
-            composable(route = Screen.Agenda.route) {
+            composable(route = Screen.Agenda.route + "/{email}",
+                arguments = listOf(
+                    navArgument("email") { type = NavType.StringType }
+                )
+            ) { backStackEntry ->
+                val email: String = backStackEntry.arguments?.getString("email") ?: ""
                 val agendaViewModel = hiltViewModel<AgendaViewModel>()
                 val state = agendaViewModel.state
+                if (email != null) {
+                    agendaViewModel.setUserInitials(email)
+                }
```

Comment:
This is a side-effect in Compose, since you execute non-Compose code (`agendaViewModel.setUserInitials()`) in a Composable context. That's not ideal, since you can't control when and how often a Compose function is called (that's handled by the Compose runtime). Having such non-Compose functions in there could execute them more often than you want to.
In this specific case it's unlikely going to be an issue, but imagine this call to the VM would fire a network request. Then every time, there's a recomposition (potentially many times a second if you're running an animation), you'd fire another network request.

Instead, you can get the email argument directly in your ViewModel with SavedStateHandle:
```
@HiltViewModel
class AgendaViewModel @Inject constructor(
    // ...
    savedStateHandle: SavedStateHandle // Will be injected automatically by Hilt
): ViewModel() {

    private val email = savedStateHandle.get<String>("email")
}
```

Reply:
Do I need to put these navigation arguments somewhere or does saveStateHandle take care of all of that? Also, I used that to get the email but I can't call my viewmodel method from the viewmodel unless I put It in the onEvent function. I can't call it from a composable so how do I launch the method to create the initials?

Reply:
You don't need to put it there, that's all handled by the navigation framework already